src/python/platforms/android/adb.py

```diff
@@ -419,11 +435,31 @@ def reset_device_connection():
   state = get_device_state()
   if state != 'device':
     logs.log_warn('Device state is %s, unable to recover using usb reset/'
-                  'gce reconnect.' % str(state))
+                  'cuttlefish reconnect.' % str(state))
     return False
 
   return True
 
+def get_cuttlefish_device_ip():
```

Comment:
These all need docstrings to describe these functions.

Please run `python butler.py lint` and `python butler.py format` to resolve many formatting/linting issues in your PR.